pkg/retrieval/retrieval.go

```diff
@@ -108,51 +126,83 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address) (swarm.
 		var (
 			peerAttempt  int
 			peersResults int
-			resultC      = make(chan swarm.Chunk, maxPeers)
-			errC         = make(chan error, maxPeers)
+			resultC      = make(chan retrievalResult, maxSelects)
 		)
 
-		for {
-			if peerAttempt < maxPeers {
-				peerAttempt++
+		requestAttempt := 0
 
-				s.metrics.PeerRequestCounter.Inc()
+		lastTime := time.Now().Unix()
 
-				go func() {
-					chunk, peer, err := s.retrieveChunk(ctx, addr, sp)
-					if err != nil {
-						if !peer.IsZero() {
-							logger.Debugf("retrieval: failed to get chunk %s from peer %s: %v", addr, peer, err)
-						}
+		for requestAttempt < 5 {
```

Comment:
maybe we should stay consistent and have this as a variable at the top of the file